cmds/switch_root/switch_root.go

```diff
@@ -0,0 +1,190 @@
+// Copyright 2012-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"syscall"
+)
+
+var (
+	help    = flag.Bool("h", false, "Help")
+	version = flag.Bool("V", false, "Version")
+)
+
+// Return the usage string
+func usage() string {
+	return "switch_root [-h] [-V]\nswitch_root newroot init"
+}
+
+// Recursively deletes everything at slash
+// Does not continue down other filesystems i.e.
+// new_root, devtmpfs, profs and sysfs
+func littleDoctor(path string, fs *syscall.Statfs_t) error {
```

Comment:
Busybox's [switch_root](https://github.com/mozilla-b2g/busybox/blob/master/util-linux/switch_root.c) and Klibc's [run_init](https://kernel.googlesource.com/pub/scm/libs/klibc/klibc/+/klibc-0.130/utils/run-init.c#127) both had different names, so I opted for Klibc's funnier take for a recursive deletion.

It's a reference to [Ender's Game](http://enderverse.wikia.com/wiki/Molecular_Disruption_Device).